intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordApi.kt

```diff
@@ -109,76 +127,83 @@ object MirrordApi {
         configFile: String?,
         project: Project?,
         wslDistribution: WSLDistribution?
-    ): MutableMap<String, String> {
-        val commandLine = GeneralCommandLine(cliPath(wslDistribution), "ext")
-
-        target?.let {
-            commandLine.addParameter("-t")
-            commandLine.addParameter(target)
-        }
+    ): MutableMap<String, String>? {
+        val commandLine = GeneralCommandLine(cliPath(wslDistribution), "ext").apply {
+            target?.let {
+                addParameter("-t")
+                addParameter(it)
+            }
 
-        configFile?.let {
-            // Format according to wslDistribution if exists
-            val formattedPath = wslDistribution?.getWslPath(it) ?: it
-            commandLine.addParameter("-f")
-            commandLine.addParameter(formattedPath)
-        }
+            configFile?.let {
+                val formattedPath = wslDistribution?.getWslPath(it) ?: it
+                addParameter("-f")
+                addParameter(formattedPath)
+            }
 
-        wslDistribution?.let {
-            val wslOptions = WSLCommandLineOptions()
-            wslOptions.isLaunchWithWslExe = true
-            it.patchCommandLine(commandLine, project, wslOptions)
+            wslDistribution?.let {
+                val wslOptions = WSLCommandLineOptions().apply {
+                    isLaunchWithWslExe = true
+                }
+                it.patchCommandLine(this, project, wslOptions)
+            }
         }
 
-        logger.info("running mirrord with following commandline: %s".format(commandLine.commandLineString))
+        logger.info("running mirrord with following command line: ${commandLine.commandLineString}")
 
         val process = commandLine.toProcessBuilder()
             .redirectOutput(ProcessBuilder.Redirect.PIPE)
             .redirectError(ProcessBuilder.Redirect.PIPE)
             .start()
 
+        logger.debug("waiting for process to finish")
+        val exitStatus = process.waitFor(15, TimeUnit.SECONDS)
```

Comment:
What does this timeout mean for UX? Is there a situation where the user waits for 15 seconds?